pkg/models/object.go

```diff
@@ -15,23 +15,36 @@ const (
 	// Types: influx, prometheus, testdata, ...
 	StandardKindDataSource = "ds"
 
-	// StandardKindPanel: currently used in two ways :( in search it defines a panel within a dashboard
-	// This is also used to refer to a panel plugin with type:  heatmap, timeseries, table, ...
+	// StandardKindPanel: only used for searchV2 right now
+	// Standalone panel is not an object kind yet -- library panel, or nested in dashboard
 	StandardKindPanel = "panel"
 
-	// StandardKindTransform: used to show that a dashboard depends on a set of transformations
-	// NOTE: this should likey be replaced with kind:system/type:transform/uid:joinByField or something like that
-	StandardKindTransform = "transform"
-
 	// StandardKindSVG SVG file support
 	StandardKindSVG = "svg"
 
 	// StandardKindPNG PNG file support
 	StandardKindPNG = "png"
 
+	// StandardKindGeoJSON represents spatial data
+	StandardKindGeoJSON = "geojson"
+
 	// StandardKindQuery early development on panel query library
 	// the kind may need to change to better encapsulate { targets:[], transforms:[] }
 	StandardKindQuery = "query"
+
+	//----------------------------------------
+	// References are referenced from objects
+	//----------------------------------------
+
```

Comment:
so iiuc, the `StandardReferenceX` is used for entities that:
- are referred to by entities stored in the object store
- are not stored in the object store themselves

I like this distinction - object store internal will have to treat these two types of references differently (e.g. graph resolution algorithm can't process the "external" refs at all).


I don't get the distinction between `StandardReferencePlugin` and `StandardReferenceType_RuntimeTransformer` - to me they look the same, and both should be on the same level, ie. either

```
- ds
- dashboard
- query
- png
- runtime
├── plugin
└── transformer
```

or..



```
- ds
- dashboard
- query
- png
- ext_plugin / runtime_plugin
- ext_transformer / runtime_transformer
```

Also - would we want to reuse those 'runtime' references for non-grafana entites? E.g. google cal, maps, sso team etc

Reply:
I agree the taxonomy here is challenging -- but think this is progress in the right direction.

"runtime" is not great -- maybe "frontend"?  

plugin is big enough and has a clear 'type'

Flat seems attractive, but the list will get pretty long and has implied heiarchy anyway:

- panel-plugin
- datasource-plugin
- app-plugin
- influx-ds.  (datasource instance of type influx -- also requires the influx datasource plugin!)
- prometheus-ds
- ...



`StandardReferenceType_RuntimeTransformer` -- mainly a constant for transformer.  could/should move to a registry of the possible frontend types 🤷

Reply:
I am not sure about `frontend` - I am pretty sure we could use this pattern for backend/non-grafana entities as well (like plugin?). Wdyt about `ExternalEntityReferenceX`, as in external to the object store/entity api? I think that's the most important distinction for us